test/lib.sh

```diff
@@ -295,3 +295,31 @@ last_commit_message() {
         git log -1 | tail -n +5
     )
 }
+
+## expect_failure ERROR_STRING CMD ARGS...
+#
+# Runs CMD with ARGS...
+# Fails if CMD succeeds.
+# If CMD fails, the output (stdout+stderr) is grepped for ERROR_STRING,
+# and we succeed iff it is found.
+# CMD ARGS... is run via eval "$@". Quote accordingly.
+expect_failure () {
+  local errstr=$1
```

Comment:
nit: seems we are at 4 spaces here for indentation

Reply:
Ack, will fix. I should just set my vim default...

Reply:
Done.